test/widgets/action_sheet_test.dart

```diff
@@ -1188,6 +1189,199 @@ void main() {
+      await tester.pump();
+      check(find.byType(SubscriptionListPageBody)).findsOne();
+
+      await tester.longPress(find.text(someChannel.name).hitTestable());
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    Future<void> showFromAppBar(WidgetTester tester, {
+      ZulipStream? channel,
+      String topic = someTopic,
+      List<StreamMessage>? messages,
+      bool isTopicNarrow = false,
+    }) async {
+      channel ??= someChannel;
+      messages ??= [someMessage];
+
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: messages).toJson());
+      await tester.pumpWidget(TestZulipApp(
+        accountId: eg.selfAccount.id,
+        child: MessageListPage(
+          initNarrow: isTopicNarrow
+            ? TopicNarrow(channel.streamId, TopicName(topic))
+            : ChannelNarrow(channel.streamId))));
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.descendant(
+        of: find.byType(ZulipAppBar),
+        matching: find.text(channel.name)));
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    Future<void> showFromRecipientHeader(WidgetTester tester, {
+      StreamMessage? message,
+    }) async {
+      message ??= someMessage;
+
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [message]).toJson());
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const MessageListPage(initNarrow: CombinedFeedNarrow())));
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.descendant(
+        of: find.byType(RecipientHeader),
+        matching: find.text(message.displayRecipient ?? '')));
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    final actionSheetFinder = find.byType(BottomSheet);
+    Finder findButtonForLabel(String label) =>
+      find.descendant(of: actionSheetFinder, matching: find.text(label));
+
+    void checkButton(String label) {
+      check(findButtonForLabel(label)).findsOne();
+    }
+
+    group('showChannelActionSheet', () {
+      void checkButtons() {
+        check(actionSheetFinder).findsOne();
+        checkButton('Mark channel as read');
+      }
+
+      testWidgets('show from inbox', (tester) async {
+        await prepare();
+        await showFromInbox(tester);
+        checkButtons();
+      });
+
+      testWidgets('show from subscription list', (tester) async {
+        await prepare();
+        await showFromSubscriptionList(tester);
+        checkButtons();
+      });
+
+      testWidgets('show with no unread messages', (tester) async {
+        await prepare(unreadMsgs: eg.unreadMsgs());
+        await showFromSubscriptionList(tester);
+        check(actionSheetFinder).findsNothing();
+      });
+
+      testWidgets('show from app bar in channel narrow', (tester) async {
+        await prepare();
+        await showFromAppBar(tester);
```

Comment:
Each test should make explicit the key test data that's needed in order for the test to make sense.

So if this is testing what happens in a channel narrow (not just some arbitrary narrow where it doesn't matter which one), it should specify a channel narrow:
```suggestion
      testWidgets('show from app bar in channel narrow', (tester) async {
        final narrow = ChannelNarrow(/* … */);
        await prepare();
        await showFromAppBar(tester, narrow: narrow);
```